Back/Reciplastk.Gateway/Services/ShipmentService.cs

```diff
@@ -0,0 +1,114 @@
+        {
+            var shipment = db.Shipments.Where(x => x.Shipmentid == shipmentid).FirstOrDefault();
+            return shipment;
+        }
+        public HttpResponseModel CreateShipment(ShipmentViewModel shipmentViewModel)
+        {
+            var response = new HttpResponseModel();
+            var shipment = FindShipmentById(shipmentViewModel.shipmentid);
+            if (shipment == null)
+            {
+                var newShipment = new Shipment();
+                newShipment.Customerid = shipmentViewModel.cutomerid;
+                newShipment.Employeeid = shipmentViewModel.employyeid;
+                newShipment.Shipmenttypeid = shipmentViewModel.shipmenttypeid;
+                newShipment.Shipmentstartdate = DateTime.Now;
+                newShipment.Shipmentstartend = DateTime.Now;
+                newShipment.Ispaid = shipmentViewModel.ispaid;
+                newShipment.Iscomplete = shipmentViewModel.iscomplete;
+                newShipment.Isactive = shipmentViewModel.isactive;
+                db.Shipments.Add(newShipment);
+                db.SaveChanges();
+                response.WasSuccessful = true;
+                response.Data = newShipment;
+                response.StatusMessage = "El cargamento se creo exitosamnete";
+            }
+            else
+            {
+                response.WasSuccessful = false;
+                response.StatusMessage = "Ya existe otro cargamento con el id indicado";
+            }
+            return response;
+        }
+        public HttpResponseModel EditShipment(ShipmentViewModel shipmentViewModel)
+        {
+            var response = new HttpResponseModel();
+            var shipment = FindShipmentById(shipmentViewModel.shipmentid);
+            if (shipment != null) 
+            {
+                shipment.Customerid = shipmentViewModel.cutomerid;
+                shipment.Employeeid = shipmentViewModel.employyeid;
+                shipment.Shipmenttypeid = shipmentViewModel.shipmenttypeid;
+                shipment.Ispaid = shipmentViewModel.ispaid;
+                shipment.Iscomplete = shipmentViewModel.iscomplete;
+                shipment.Isactive = shipmentViewModel.isactive;
+                db.Shipments.Add(shipment);
+                db.SaveChanges();
+                response.WasSuccessful = true;
+                response.Data = shipment;
+                response.StatusMessage = "El cargamento se edito exitosamnete";
+            }
+            else
+            {
+                response.WasSuccessful = false;
+                response.StatusMessage = "No se encontro ningun cargamento con el id indicado";
+            }
+            return response;
+        }
+        public HttpResponseModel DeleteShipment(int shipmentid)
+        {
+            var response = new HttpResponseModel();
+            var shipment = FindShipmentById(shipmentid);
+            if (shipment != null)
+            {
+                db.Remove(shipment);
```

Comment:
Use Soft delete instead.
```suggestion
                shipment.IsActive = false;
```